content/quote-pattern-type-variable-syntax.md

```diff
@@ -0,0 +1,139 @@
+```scala
+case '{ type t; $x: `t` } => f[t](x: Expr[t])
+case '{ type u; ($ls: List[`u`]).map($f: `u` => Int) } => g[u](ls: Expr[List[u]], f: Expr[u => Int])
+case '{ type tail <: Tuple; $x: *[Int, `tail`] } => h[tail](x: Expr[*[Int, tail])
+```
+
+##### Nested type variable
+Types with a type variable name introduce a new type variable. These cannot be references with a backticked reference due to their scope. We cannot add explicit bounds to them, but in certain cases we can infer their some bounds. These variables become explicit type variables in the internal representation after typing.
+```scala
+case '{ $x: t } => f[t](x: Expr[t])
+```
+
+
+##### Type Patterns
+Quoted type patterns only support nested type variable definitions. Explicit type variables are not supported in the source due to an oversight. These variables become explicit type variables in the internal representation after typing. The bounds of the type variable are `Any` and `Nothing`.
+```scala
+case '[ t ] => f[t]
+case '[ List[t] ] => g[t]
+```
+
+### Support type bounds in quoted type patterns
+
+We want to be able to set the bounds of type variables to be able to match against type constructors that have type bounds. For example, the tuple `*:` type.
+```scala
+case '[ head *: tail ] => h[tail]
+```
+See https://github.com/lampepfl/dotty/issues/11738.
+
+### Support matching on any kind of type
+We want to match against higher-kinded (or `AnyKind`) types. This is not possible due to the default upper bound of `Any`.
+See https://github.com/lampepfl/dotty/issues/10864.
+
+### Support multiple references to the same type in quoted type patterns
+We want to be able to match using several references to the same type variable.
+```scala
+case '[ (t, t, t) ] => f[t] // t is going to match the glb of the tuple T1, T2, T3
+```
+
+### Simplify the use of explicit type variables
+It is inconsistent to need to use backticks for references to explicit type variables in the quote but not outside.
+We want to be able to refer to the variable by its non-backticked name uniformly.
+```diff
+- case '{ type u; ($ls: List[`u`]).map($f: `u` => Int) } => g[u](ls: Expr[List[u]], f: Expr[u => Int])
++ case '{ type u; ($ls: List[u]).map($f: u => Int) } => g[u](ls: Expr[List[u]], f: Expr[u => Int])
+```
+
+## Proposed solution
+
+### High-level overview
+
+We first want to introduce syntax for explicit type variable definitions in quoted type patterns that aligns with expression quoted patterns. We can use the syntax described in [explicit type variables](Explicit-type-variables).
+
+```scala
+case '[ type t; List[`t`] ] => f[t]
+case '[ type tail <: Tuple <: *[Int, `tail`] ] => g[tail]
```

Comment:
It looks like the typo was fixed in another place in the file but not here

Reply:
Fixed